### ISSUE TITLE
Multiple OSC Send Targets

### ISSUE THREAD

**Issue by davidmw** [monsieurgustav/UE4-OSC#16]
I'm having trouble sending to multiple targets.  I have three iPads added to the OSC plugin config screen (192.168.0.40:8000, 192.168.0.41:8000, 192.168.0.42:8000). Whichever IP is in the first slot is the iPad that seems to get all the messaging from the plugin.

When there are multiple send targets possible it seems necessary to allow selection of a specific send target at run time. I want to acknowledge input on any specific iPad with a OSC message back to that iPad. So at times I want to broadcast to all send targets and at other times would like to be selective. Seems like the node should make the inbound IP or an index into the send target array available so we know which device to reply to. I'm using port 8000 for all the iPads and 9000 on the plugin - it would be very nice to avoid building and parsing IP:Port addresses in blueprints to make this all happen.

**Comment by robdeja** [comment]
I'm also having that issue with multiple targets. 

I have a situation where I'm also connecting with ipads over wifi but I have variable numbers of ipads connecting - sometimes just 1 sometimes 5 (max).

I would like to be able to add target IPs (and ports) via blueprint to have a little more flexibility but I also agree that being able to tell what IP the request was received on would be perfect as I could then "reply" to that device.

**Comment by monsieurgustav** [comment]
About the "reply" problem:
I'm not really willing to add an IP parameter to the sendOsc function. The
ability to manage IP globally in the settings, possibly per deployment, has
proven useful to me.
But the reply pattern is essential...

A possible solution ?
- add a AddDestIP function, that takes a string and returns an integer to
feed into the SendOsc function, and
- for every received message, the sender is automatically added to the IPs
list, and the OnOscReceived event comes with the index of the sender that
can be used in the SendOsc function.


2015-12-08 10:51 GMT+01:00 Rob Deja <notifications@github.com>:

**Comment by robdeja** [comment]
Thanks. That sounds like it would be a reasonable solution.

**Comment by davidmw** [comment]
Agreed - proposed solution seems to address the issues.

**Comment by monsieurgustav** [comment]
Some news about this issue:
- this commit (https://github.com/monsieurgustav/UE4-OSC/commit/b33fc3bfdf224c4751711430bdb26b6b3a64e918) adds a BP function to add a new send target at runtime.

- multiple send target seems to work:
  * TargetIndex == -1 -> send to all targets
  * TargetIndex >= 0 -> send to the ith target only

  I should update the doc on this feature.

- I implemented a first version of the solution I suggested above (the sender is automatically added to the send targets and the event gives the TargetIndex to use to reply). Unfortunately, it doesn't really work...
The source IP is right, but the source port is the port of the sender (generally a random port, e.g. "30126") not the unknown listener.
